packages/rainbowkit/src/wallets/getWalletConnectConnector.ts

```diff
@@ -1,67 +1,106 @@
-}
+// Function to get or create a walletConnect instance
+const getOrCreateWalletConnectInstance = ({
+  projectId,
+  walletConnectParameters,
+  rkDetailsShowQrModal,
+}: GetOrCreateWalletConnectInstanceParams): ReturnType<
+  typeof walletConnect
+> => {
+  let config: WalletConnectParameters = {
+    ...(walletConnectParameters ? walletConnectParameters : {}),
+    projectId,
+    showQrModal: false, // Required. Otherwise WalletConnect modal (Web3Modal) will popup during time of connection for a wallet
+  };
 
-export function getWalletConnectConnector(config: {
-  projectId: string;
-  chains: Chain[];
-  options?: WalletConnectConnectorOptions;
-}): WalletConnectConnector;
+  // `rkDetailsShowQrModal` should always be `true`
+  if (rkDetailsShowQrModal) {
+    config = { ...config, showQrModal: true };
+  }
+
+  const serializedConfig = JSON.stringify(config);
+
+  const sharedWalletConnector = walletConnectInstances.get(serializedConfig);
 
+  if (sharedWalletConnector) {
+    return sharedWalletConnector;
+  }
+
+  // Create a new walletConnect instance and store it
+  const newWalletConnectInstance = walletConnect(config);
+
+  walletConnectInstances.set(serializedConfig, newWalletConnectInstance);
+
+  return newWalletConnectInstance;
+};
+
+// Creates a WalletConnect connector with the given project ID and additional options.
+function createWalletConnectConnector({
+  projectId,
+  walletDetails,
+  walletConnectParameters,
+}: CreateWalletConnectConnectorParams): CreateConnectorFn {
+  // Create and configure the WalletConnect connector with project ID and options.
+  return createConnector((config) => ({
+    ...getOrCreateWalletConnectInstance({
+      projectId,
+      walletConnectParameters,
+      // Used in `connectorsForWallets` to add another
+      // walletConnect wallet into rainbowkit with modal popup option
+      rkDetailsShowQrModal: walletDetails.rkDetails.showQrModal,
+    })(config),
+    ...walletDetails,
+  }));
+}
+
+// Factory function to obtain a configured WalletConnect connector.
 export function getWalletConnectConnector({
-  chains,
-  options = {},
   projectId,
-}: {
-  chains: Chain[];
-  projectId?: string;
-  options?: WalletConnectConnectorOptions;
-}): WalletConnectConnector {
-  // We use this projectId in place of YOUR_PROJECT_ID for our examples.
-  // This allows us our examples and templates to be functional with WalletConnect v2.
-  // We warn developers against using this projectId in their dApp in production.
-  const exampleProjectId = '21fef48091f12692cad574a6f7753643';
-  if (!projectId || projectId === '')
+  walletConnectParameters,
+}: GetWalletConnectConnectorParams): CreateConnector {
+  // Check if the projectId is provided; if not, throw an error.
+  if (!projectId) {
     throw new Error(
       'No projectId found. Every dApp must now provide a WalletConnect Cloud projectId to enable WalletConnect v2 https://www.rainbowkit.com/docs/installation#configure',
     );
```

Comment:
This is causing issues with the examples; we need to include these fallback projectIds, as documented.